controllers/namespace/namespace_controller.go

```diff
@@ -83,23 +85,26 @@ func (r *ReconcileNamespaces) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, fmt.Errorf("failed to query Namespace: %w", err)
 	}
 
-	if ns.Labels == nil {
-		return reconcile.Result{}, nil
+	codeModuleDynakubes, err := codemodules.FindCodeModules(ctx, r.client)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to query CodeModule enabled Dynaekubes: %w", errors.WithStack(err))
 	}
 
-	oaName := ns.Labels[webhook.LabelInstance]
-	if oaName == "" {
-		return reconcile.Result{}, nil
+	dk, err := codemodules.MatchForNamespace(codeModuleDynakubes, &ns)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not match CodeModule with namespace: %w", errors.WithStack(err))
 	}
 
-	var apm dynatracev1alpha1.DynaKube
-	if err := r.client.Get(ctx, client.ObjectKey{Name: oaName, Namespace: r.namespace}, &apm); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to query DynaKubes: %w", err)
+	if dk == nil {
+		// No CodeModule enabled Dynakube for given namespace found
+		return reconcile.Result{}, nil
 	}
 
-	tokenName := utils.GetTokensName(&apm)
-	if !apm.Spec.CodeModules.Enabled {
-		r.ensureSecretDeleted(tokenName, targetNS)
+	tokenName := utils.GetTokensName(dk)
+	if !dk.Spec.CodeModules.Enabled {
+		// If secret does not exist, this method throws a not found error
+		// Suppress the error in that case
+		_ = r.ensureSecretDeleted(tokenName, targetNS)
```

Comment:
Is this comment even true? According to the implementation this should actually be not the case.

Reply:
My mistake, I saw that the error was ignored (not suppressed) before and thought it was because of a potential not found error, because I overlooked `!k8serrors.IsNotFound(err)`  in the condition. I'll remove the comment, but I wonder why the error is ignored in the first place